extension/src/extension.ts

```diff
@@ -443,9 +442,26 @@ export class Extension extends Disposable implements IExtension {
     this.status.setAvailability(available)
     this.setCommandsAvailability(available)
     this.cliAccessible = available
+    this.getStarted.sendDataToWebview()
     return available
   }
 
+  public getAvailable() {
+    return this.cliAccessible
+  }
+
+  public async initProject() {
+    const root = getFirstWorkspaceFolder()
+    if (root) {
+      await this.dvcExecutor.init(root)
```

Comment:
[Q] I assume the reason for `willRecheck` is that we may need to wait a little bit after initialization before checking to see if the workspace is ready to go. Could we move the delay out here with an explanatory variable? Would that make the code more obvious? Or are you trying to make the UX seamless in that the user doesn't necessarily have to wait?

I think it would be ok to make everyone wait as long as we displayed a progress notification (see `Toast.showProgress`) that overall lastest for ~5 seconds. 

E.g => "Creating Project" => "Verifying Contents" => "Notifying Extension". After writing this I am not thinking that maybe we could actually check for the existence of the `.dvc` folder every 500ms. Once it is in place we could then fire `this.workspaceChanged.fire()`. That might actually simplify everything.

LMK what you think, feel free to disagree, ask clarifying questions, change no or follow up. Up to you.

Reply:
`willRecheck` is not for the after initialization. It's to check whether DVC is installed or not. If the project has a virtual environment, the watchers are going to catch when DVC gets installed later. If on setup DVC is not installed and you globally install it later, the extension won't be aware of it, unless we check every 5 seconds for it. It's inside it's own function so that it's easily mockable for the tests.